rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java

```diff
@@ -509,16 +519,16 @@ final class DataCacheEntry extends CacheEntry {
     private final RouteLookupResponse response;
     private final long expireTime;
     private final long staleTime;
-    private final ChildPolicyWrapper childPolicyWrapper;
+    private final List<ChildPolicyWrapper> childPolicyWrappers;
 
     // GuardedBy CachingRlsLbClient.lock
     DataCacheEntry(RouteLookupRequest request, final RouteLookupResponse response) {
       super(request);
       this.response = checkNotNull(response, "response");
-      // TODO(creamsoup) fallback to other targets if first one is not available
-      childPolicyWrapper =
+      checkState(!response.targets().isEmpty(), "No targets returned by RLS");
```

Comment:
I'd feel much more comfortable if this were a checked exception. Unhandled exception in LB policies put the channel in panic mode which fails all RPCs on the channel for the remainder of its life.